libraries/botbuilder-core-extensions/tests/transcriptUtilities.js

```diff
@@ -39,7 +54,161 @@ function getActivitiesFromChat(chatFilePath) {
+ * @param {function} botLogicFactoryFun Function which accepts conversationState and userState and should return the bots logic to test.
+ * @param {function} middlewareRegistrationFun (Optional) Function which accepts the testAdapter, conversationState and userState.
+ */
+function assertBotLogicWithBotBuilderTranscript(relativeTranscriptPath, botLogicFactoryFun, middlewareRegistrationFun) {
+    return function (mochaDoneCallback) {
+
+        checkTranscriptResourcesExist()
+            .then(transcriptsBasePath => {
+                var transcriptPath = path.join(transcriptsBasePath, relativeTranscriptPath);
+                assertBotLogicWithTranscript(transcriptPath, botLogicFactoryFun, middlewareRegistrationFun)(mochaDoneCallback)
+            }).catch(mochaDoneCallback);
+    }
+}
+
+/**
+ * Creates a Mocha Test definition (Mocha.ITestDefinition) that will use the TestAdapter to test a bot logic against the specified transcript file.
+ * Optionally, pass a third parameter (as function) to register middleware into the TestAdapter.
+ * @param {string} transcriptPath Path to the transcript file. Can be a .chat or .transcript file.
+ * @param {function} botLogicFactoryFun Function which accepts conversationState and userState and should return the bots logic to test.
+ * @param {function} middlewareRegistrationFun (Optional) Function which accepts the testAdapter, conversationState and userState.
+ */
+function assertBotLogicWithTranscript(transcriptPath, botLogicFactoryFun, middlewareRegistrationFun) {
+    var loadFun = transcriptPath.endsWith('.chat')
+        ? getActivitiesFromChat
+        : getActivitiesFromTranscript;
+
+    // return a Mocha Test Definition, which accepts the done callback to indicate success or error
+    return function (done) {
+        loadFun(transcriptPath).then(activities => {
+
+            // State
+            const storage = new MemoryStorage();
+            const conversationState = new ConversationState(storage);
+            const userState = new UserState(storage);
+            const state = new BotStateSet(conversationState, userState);
+
+            // Bot logic + adapter
+            var botLogic = botLogicFactoryFun(conversationState, userState)
+            var adapter = new TestAdapter(botLogic);
+            adapter.use(state);
+
+            // Middleware registration
+            if (typeof middlewareRegistrationFun === 'function') {
+                middlewareRegistrationFun(adapter, conversationState, userState);
+            }
+
+            // Assert chain of activities
+            return adapter.testActivities(activities)
+                .then(done)
+                .catch(done);
+
+        }).catch(done);
+    }
+}
+
+
+// **** PRIVATE **** //
+
+function checkTranscriptResourcesExist() {
+    var transcriptsLocation = process.env['BOTBUILDER_TRANSCRIPTS_LOCATION'] || 'https://github.com/southworkscom/BotBuilder/archive/botbuilder-v4-transcripts.zip';
```

Comment:
Once transcripts are merged (https://github.com/Microsoft/BotBuilder/pull/4673), this URL should be replaced with:
https://github.com/Microsoft/BotBuilder/archive/master.zip